galley/pkg/config/analysis/diag/messages_test.go

```diff
@@ -0,0 +1,91 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package diag
+
+import (
+	"testing"
+
+	. "github.com/onsi/gomega"
+)
+
+func TestMessages_Sort(t *testing.T) {
+	g := NewGomegaWithT(t)
+	var msgs Messages
+	firstMsg := NewMessage(
+		NewMessageType(Error, "B1", "Template: %q"),
+		testOrigin("B"),
+		"B",
+	)
+	secondMsg := NewMessage(
+		NewMessageType(Warning, "A1", "Template: %q"),
+		testOrigin("B"),
+		"B",
+	)
+	thirdMsg := NewMessage(
+		NewMessageType(Warning, "B1", "Template: %q"),
+		testOrigin("A"),
+		"B",
+	)
+	fourthMsg := NewMessage(
+		NewMessageType(Warning, "B1", "Template: %q"),
+		testOrigin("B"),
+		"A",
+	)
+	fifthMsg := NewMessage(
+		NewMessageType(Warning, "B1", "Template: %q"),
+		testOrigin("B"),
+		"B",
+	)
+	msgs.Add(fifthMsg)
+	msgs.Add(fourthMsg)
+	msgs.Add(thirdMsg)
+	msgs.Add(secondMsg)
+	msgs.Add(firstMsg)
```

Comment:
Again, since this is very local test, I think it's fine to use the internals. This lets us express this more concisely:

```go
msgs := Messages{firstMsg, secondMsg, thirdMsg, fourthMsg, fifthMsg}
```

In fact, since Gomega is fairly smart about equality, I believe you can even express both the initial list and your expected list in this way, and compare them directly:

```go
msgs := Messages{fifthMsg, fourthMsg, thirdMsg, secondMsg, firstMsg}
expected := Messages{firstMsg, secondMsg, thirdMsg, fourthMsg, fifthMsg}
msgs.Sort()
g.Expect(msgs).To(Equal(expected))
```

The same goes for the other test, of course.